File: api_service/src/services/auth/token_svc.py
from src.models.auth.auth import IssuedToken, TokenBundle, TokenVerificationResult
from src.models.auth.auth_contract import (
    TokenRefreshRequest,
    TokenRevokeRequest,
    TokenVerifyRequest,
)


class TokenService:
    """令牌服务。运行期不回源认证中心。"""

    def __init__(self) -> None:
        pass

    def set_bootstrap_tokens(self, bundle: TokenBundle | None) -> None:
        # 非认证中心模块不再缓存 bootstrap 令牌。
        _ = bundle

    async def set_bootstrap_tokens_async(self, bundle: TokenBundle | None) -> None:
        _ = bundle

    async def get_access_token(self) -> IssuedToken | None:
        # 访问令牌的获取与轮换统一由认证中心处理；本地不维护状态。
        return None

    async def refresh(self, req: TokenRefreshRequest) -> TokenBundle | None:
        _ = req
        return None

    async def verify(self, req: TokenVerifyRequest) -> TokenVerificationResult | None:
        _ = req
        return None

    async def revoke(self, req: TokenRevokeRequest) -> None:
        _ = req

File: certification_server/src/interfaces/auth/token_svc.go
package auth

import (
	authmodel "certification_server/src/models/auth"
	"context"
)

// ITokenService 定义令牌签发、刷新、校验与撤销接口。
type ITokenService interface {
	// IssueToken 根据认证结果颁发访问令牌，返回令牌信息。
	IssueToken(ctx context.Context, req *authmodel.TokenIssueRequest) (*authmodel.IssuedToken, error)
	// IssueTokenBundle 根据会话信息颁发访问令牌，返回令牌信息和相关元数据。
	IssueTokenBundle(ctx context.Context, session *authmodel.Session, req *authmodel.TokenIssueRequest) (*authmodel.TokenBundle, error)

	// RefreshToken 根据刷新请求刷新访问令牌，必要时轮换刷新令牌，返回新的令牌信息。
	RefreshTokenBundle(ctx context.Context, req *authmodel.TokenRefreshRequest) (*authmodel.TokenBundle, error)
	// VerifyToken 校验访问令牌的有效性和权限，返回认证上下文。
	VerifyToken(ctx context.Context, req *authmodel.TokenVerifyRequest) (*authmodel.TokenVerificationResult, error)

	// RevokeToken 撤销访问令牌，更新令牌状态以拒绝后续使用。
	RevokeToken(ctx context.Context, req *authmodel.TokenRevokeRequest) error
	// RevokeTokenFamily 根据令牌族标识撤销相关令牌，通常用于强制用户下线。
	RevokeTokenFamily(ctx context.Context, familyID string, revokedBy string) error
}
